lib/spotlight/dor/indexer.rb

```diff
@@ -86,18 +86,18 @@ def add_folder(sdb, solr_doc)
       #   _location.physicalLocation should find top level and relatedItem
       folder_num = sdb.smods_rec._location.physicalLocation.map do |node|
         val = node.text
-        # folder may be text with commas
-        match_data = val.match(/Folder:? ?(.+)/i)
-        next if match_data.blank?
-        result = match_data[1].strip
-        # Menuez collection may have folder followed by Sleeve then Frame
-        match2_data = result.match(/(.*),? ?Sleeve/i)
-        if match2_data
-          match2_data[1].strip.sub(/,$/, '')
-        else
-          result
-        end
+
+        match_data = if val =~ /\|/
+                       # we assume the data is pipe-delimited, and may contain commas within values
+                       val.match(/Folder ?:? ?([^|]+)/)
+                     else
+                       # the data should be comma-delimited, and may not contain commas within values
+                       val.match(/Folder ?:? ?([^,]+)/)
+                     end
+
+        match_data[1].strip if match_data.present?
       end
+
```

Comment:
with the comments, the above makes a lot more sense to me.  Or maybe it's just because #13 is finally merged.  

The built in assumption of my approach was "folder is the last thing" -- because it was in all the cases I could find, except Menuez.  Your case has a built in assumption of properly delimited fields, and the comments explain that.